### DISCUSSION TITLE
cracked the code to get rid of artifacts

### DISCUSSION THREAD

**Discussion by loawizard** [nagadomi/nunif#114] [Ideas]
today a friend send me this clip https://drive.google.com/file/d/1gDcHS_EJ-NUdgrh6wa95R0HLglh7Epdy/view made in owl.

they figured out how to not have artifacts apparently. Maybe you can take a look at there code and backwards engineer what they did.  

As i don't want to use owl, i want to use IW3

**Comment by kaelsonofkrypto** [comment]
Yes, it looks very promising.

**Comment by donrikk** [comment]
here is an example of how well temporal stability helps edges and clean artifacts and instability in the maps and it doesnt increase wait time by to much at all maybe a extra 2 minutes on the conversion time. and temporal stability is different from temporal coherence just to clarify [https://drive.google.com/file/d/1WCT1XOrO6mDPCKVKxgkbdmZD5WvET2Zx/view?usp=sharing](url)

these settings i used in owl3D mimmick what 3 depth would relatively lok like in your program but with temporal stability on low.

**Reply by donrikk** [reply]
u can download the video from that link and view it urself with any medium really. it is remarkable and i believe it is absolutely something you could achieve. my suggestion is all the other suggestions should be put on the back burner and instead focus on this, because this seems to make 3 depth possible with minimal artifacts which is exactly what all of us and im sure you are looking for out of our conversions. so yeah ide say this would be a good thing to put focus into as this seems to make the biggest difference so far in 3d visual quality. i hope this isnt to hard to implement, i dont want to burn you out.

**Reply by loawizard** [reply]
hmmnn what's the difference between temporal stability and temoral coherence.

i do see the lack of artifacting although i do prefere iw3 feeling of depth. would be great if we could somehow combine it

**Reply by donrikk** [reply]
so temporal coherence kinda blurs the map a bit to handle artifacts in motion and so on but temporal stability has to do wit hthe stability of the depth map as a whole. it can technically decrease 3D quality at high settings but at low it stabilizes the map so u dont get tearing in backgrounds during scenes and u dont get weird pop in and pop out when more than 1 foreground object is exiting or entering the foreground. i find at low it fixes all those instability issues but doesnt sacrifice really any 3D quality as u can see from the example video shared above. if iwere to do this same conversion at 3 depth in iw3 currently ill get a massive amount of haloing and background tearing. but with temporal stability implimented that would fix those issues and bring the program more towards studio level 3D. this is a good middle ground until that method released by disney is more viable for consumers or at the least trickled down to github for open source application if disney is feeling generous and dont patent it which they probably will lol

**Reply by donrikk** [reply]
basically temporal stability mixed with flip augmentation or TTA in iw3 and default to 512 map resolution and the artifacts should be stiffled in iw3 u have all the puzzle pieces except this last one.

**Comment by loawizard** [comment]
> u can download the video from that link and view it urself with any medium really. it is remarkable and i believe it is absolutely something you could achieve. my suggestion is all the other suggestions should be put on the back burner and instead focus on this, because this seems to make 3 depth possible with minimal artifacts which is exactly what all of us and im sure you are looking for out of our conversions. so yeah ide say this would be a good thing to put focus into as this seems to make the biggest difference so far in 3d visual quality. i hope this isnt to hard to implement, i dont want to burn you out.

yep indeed. the biggest workingpoint of iw3 is the artifacting i think. if this can remove that , then all other problems resolve by itself.

**Comment by loawizard** [comment]
One thing i do have to say and some don't agree with that. I think IW3 is miles ahead when it comes to depth. Depth at 2.5 any b , 0.5 convergence and own size 3 is just amazing. It's like your in the clip. Feels very natural to me. would be great to see if we can figure out what owl3D is doing for the artifacting to make the best converter available.  i think it might be the temporal stability

**Comment by loawizard** [comment]
This is what a friend shared with me

midas 3 is better trained on cinema and produces much more accurate cinema like results and midas 3 mixed with TTA and temporal stability even with lower depth resolution maps will achieve a stable clean 3D experience at 2.5 depth and higher with minimal artifacts on final result. i would suggest looking into and adding midas 3 as a depth model option. this is the github https:// github.com/isl-org/MiDaS

For the latest release MiDaS 3.1, a technical report and video are available.
"MiDaS was trained on up to 12 datasets (ReDWeb, DIML, Movies, MegaDepth, WSVD, TartanAir, HRWSI, ApolloScape, BlendedMVS, IRS, KITTI, NYU Depth V2) with multi-objective optimization."

 it has weights and stuff just like depth anything model and zoedepth and all other models that use monocular depth estimation so it should be as easy if not similar to implement that depth map model as all the others that have been added.

if this map and temporal stability is added then ull have pretty much near studio results if not better becuase the program dual renders and studios only single eye render. so your program can potentially produce better products than Hollywood

This has an open source MIT license and is completely ready to implement with no license restrictions!

Although me personally I prefer depth of anylarge  some iw3 users told me they would want this option.

I'm just keen to know if we can have temporal stability on the existing depthmaps.

**Reply by donrikk** [reply]
i would love to also use temporal stability with depth anything aswell as its not something i can do on an other program and it would be probably the first real produced test of temporal stability using depth anything, maybe we will strike gold!

**Reply by nagadomi** [reply]
ZoeDepth uses MiDaS 3.1 (DPT_BEiT_L_384) as its backbone.
DepthAnything outperforms MiDaS 3.1 on multiple evaluation datasets.
So I am guessing that the current MiDaS will not improve anything.

MiDaS is the most widely used depth estimation model, so it's not a bad idea to support it, but I feel it can wait until the next version is released (I don't know if it will be).

**Reply by gituser123456789000** [reply]
@nagadomi 
Adding MiDaS would be a waste in my opinion and taking a step backwards. It's outdated, as you stated. ZoeDepth builds on it with better accuracy, Depth-Anything has better accuracy. Adding MiDaS (unless they come out with an improved 4.0+ version) seems like looking backwards instead of forward.

Owl's new Ultra model appears to be an improved version of Tiled ZoeDepth_N. It pulls out the same details, but doesn't have the intense, overly-bright result of the main TiledZoeDepth and GUI fork.. and the gradient scale looks like it fades accurately, like Depth-Anything.. so the final result looks very nice in the testing I've done so far. I haven't seen any mid-range 'bulge' effect with it yet (something Depth-Anything is also correct with, but many other models fail at with some content).

I'm still very interested in seeing Tiling (or something like BoostingMonocularDepth, if that's better) added to iw3 in a similar or even further improved way.. so we could then do Tiled ZoeDepth_N without resolution and time limitations.. and also see what Tiled Depth-Anything would look like.

I saw elsewhere that you want to avoid inpainting, but that should be the next big step coming to 2D-3D conversions, for cleaning up the edge blurs and distortions, which is the last major issue that really stands out with the quality of the final results.

I don't mind waiting longer for processing if Tiling and Inpainting options produce a superior result.

I haven't been able to process full videos yet in iw3, because my computer crashes with the settings I'd like to use (I assume it's been the power supply eventually overheating). But I should have a multi-bay external hard drive dock coming soon that will free up some wattage of my power supply by not having to power some drives, producing less heat, and with the physical drives removed, I'll have much less restricted airflow for cooling. I wanted everything inside the PC case.. 1 power cord, not having extra wires everywhere.. but now I realize that a wall of solid metal hard drives between the intake fans and what they're mostly supposed to be trying to cool (CPU, GPU, power supply, ram, etc).. was probably not such a good idea.

Similar to this:
<img width="573" alt="Capture" src="https://github.com/nagadomi/nunif/assets/156575428/58b6a961-5c79-4f50-be00-46d89fe2b9cb">

**Reply by nagadomi** [reply]
I would consider improving the depth in June or later.
I am currently working on image warping(row_flow model) and will be away in May.

**Reply by loawizard** [reply]
I just want to thank you for the amazing program, Nagadomi. I use it all the time and it has created some wonderful experiences for me. Thank you so much for your work; I greatly appreciate it! Enjoy your time off in May, God knows you deserve it!

**Reply by gituser123456789000** [reply]
FYI, Pimax Crystal Super was announced yesterday and is up for preorder spot reservations $0.99 USD
3840x3840 per eye.. that's ~78% higher than the Pimax Crystal, 26% higher than the Apple Vision Pro (and about half the price), over 3x the resolution of the Reverb G2 / Quest 3 / Pico 4 ~2160x2160 tier headsets
50-57 PPD
QLED and micro-OLED display options.... or BOTH... SWAPABLE DISPLAYS. That's crazy.. so maybe future-proof when 5000x5000 displays are possible.. or 8000x8000 displays.. just buy a new display instead of a whole headset
PCVR only.. no more battery like the Crystal.. 'smaller' (idk by how much).. less weight
Wider FOV 130* vs 110* of the Crystal
https://www.youtube.com/watch?v=mZRC-oNhSYs&t=1317s

**Reply by kaelsonofkrypto** [reply]
thanks for your work on the program, enjoy the vacation
